src/main/java/com/shapesecurity/salvation/directives/AncestorSourceListDirective.java

```diff
@@ -10,4 +13,9 @@
         @Nonnull Set<AncestorSource> ancestorSources) {
         super(name, ancestorSources);
     }
+
+    public boolean matchesSource(@Nonnull Origin origin, @Nonnull URI source) {
```

Comment:
Don't you want to say `implements MatchesSource` or something before adding this?